inlong-sort/sort-formats/format-json/src/main/java/org/apache/inlong/sort/formats/json/canal/CanalJsonEnhancedDeserializationSchema.java

```diff
@@ -224,6 +236,11 @@ public void deserialize(@Nullable byte[] message, Collector<RowData> out) throws
             }
             final GenericRowData row = (GenericRowData) jsonDeserializer.convertToRowData(root);
             String type = row.getString(2).toString(); // "type" field
+            // only not appendMode type field is valid
+            if (appendMode) {
```

Comment:
This judgment should be removed to avoid the type of output being INSERT?

Reply:
avoid add option in different source format. Import hive sink changelog data, regard upsert and delete data as insert data.